### PR TITLE
Add big lru map to utils

### PR DESCRIPTION
**Pull request by jeffmontagna**
- Added new class `BigLruMap` that extends `LRUMap` to include `BigMap` and a flexible array to avoid running out of memory with 11.8 million items in the map.

Closes https://github.com/terascope/teraslice/issues/2703

### CODE REVIEW

File: packages/utils/src/big-lru-map.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+* the array in LRUMap. Doing this will avoid
+* running out of memory at 12 million records.
+* */
+class FlexibleArray {
+    constructor() {
+        return [];
+    }
+}
+
+export class BigLruMap<V> extends LRUMap<string|number, V> {
```
[the remark this answers is not in the document]
Comment (jeffmontagna) [reply packages/utils/src/big-lru-map.ts]: Changed class name to `BigLRUMap`

File: packages/utils/src/big-lru-map.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    constructor(cacheSize: number) {
```
[the remark this answers is not in the document]
Comment (jeffmontagna) [reply packages/utils/src/big-lru-map.ts]: Added support to key and value arrays being typed.

File: packages/utils/src/big-lru-map.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* running out of memory at 12 million records.
+* */
+export class FlexibleArray {
+    constructor() {
+        return [];
+    }
+}
+
+export class BigLRUMap<V> extends LRUMap<string|number, V> {
+    constructor(mapSize: number, keyArray?: FlexibleArray, valueArray?: FlexibleArray) {
```
Comment (peterdemartini) [inline packages/utils/src/big-lru-map.ts:18]: We need to make sure that the flexible array is passed in, also this should support a `TypedArray` (this comes from `@terascope/types`)

```suggestion
    constructor(mapSize: number, keyArray: FlexibleArray|TypedArray = FlexibleArray, valueArray: FlexibleArray|TypedArray = FlexibleArray) {
```

Comment (jeffmontagna) [reply packages/utils/src/big-lru-map.ts:18]: If keyArray and valueArray are not optional would it make sense to move FlexibleArray to @terascope/types?

Comment (peterdemartini) [reply packages/utils/src/big-lru-map.ts:18]: Not really, this `FlexibleArray` is really a hack and isn't a type only thing. Ideally we shouldn't even be exporting it

**Review by peterdemartini — APPROVED** [review]
LGTM

**Pull request merged**